src/Uno.UI/UI/Xaml/Controls/ScrollContentPresenter/NativeScrollContentPresenter.iOS.cs

```diff
@@ -166,6 +166,11 @@ public override void SetContentOffset(CGPoint contentOffset, bool animated)
 				previousView.RemoveFromSuperview();
 			}
 
+			while (Subviews.Length > 0)
```

Comment:
```suggestion
            // Ensure we're working with an empty view, in case previously removed views were missed.
			while (Subviews.Length > 0)
```